src/west/cmd/project.py

```diff
@@ -494,6 +494,12 @@ def _verify_repo(project):
                       "'git rev-parse --show-cdup'")
 
 
+def _exists(project):
```

Comment:
> Maybe It'd be better to have an _is_repo() function that does the _verify_repo() thing (after checking that the path exists, maybe), but returns True/False. Could do the clone if not _is_repo() then, and just call _is_repo() in _cloned().

Sure, I'll make that change.

> One reason I didn't want to test directly for .git/ is that I think there are options for giving it other names. Not that that's very likely to happen though...

We're doing the cloning though, not the user, so I think that's a "can't happen", right?

> Still not sure how you end up with the directory existing but not being a Git repository, when using Git commands, but maybe it can happen.

I tried to explain in the commit log how this can happen -- when a repository contains submodules and west is being used to manually put them into place (rather than requiring the user to invoke git submodule commands, which are icky). This is a port of a repo idiom.

It happens in practice when using mcuboot:

```
$ git clone https://github.com/runtimeco/mcuboot
$ ls -a mcuboot/sim/mcuboot-sys/mbedtls
.  ..
$ cat mcuboot/.gitmodules 
[submodule "sim/mbedtls"]
	path = sim/mcuboot-sys/mbedtls
	url = https://github.com/ARMmbed/mbedtl
```

Reply:
>> Maybe It'd be better to have an _is_repo() function that does the _verify_repo() thing (after checking that the path exists, maybe), but returns True/False.

Actually, I don't understand. That's exactly what `_cloned` already is, no?

```
def _cloned(project):
    # Returns True if the project's repository exists and looks like a Git
    # repository.
    #
    # Prints a warning if the project's clone path exist but doesn't look like
    # a Git repository.

    if _exists(project):
        _verify_repo(project)
        return True

    return False
```

>> Could do the clone if not _is_repo() then, and just call _is_repo() in _cloned().

Given that, I'm not sure what you're asking for here.

Reply:
> > One reason I didn't want to test directly for .git/ is that I think there are options for giving it other names. Not that that's very likely to happen though...
> 
> We're doing the cloning though, not the user, so I think that's a "can't happen", right?

Feels more robust to do it with `git` commands at least. Never know what situations you might run into later on.

> > Still not sure how you end up with the directory existing but not being a Git repository, when using Git commands, but maybe it can happen.
> 
> I tried to explain in the commit log how this can happen -- when a repository contains submodules and west is being used to manually put them into place (rather than requiring the user to invoke git submodule commands, which are icky). This is a port of a repo idiom.

Hmm... had no idea `git` by itself would ever create empty directories (since you can't check in empty directories, afaiu), but looks like it does it for submodules. Yeah, that makes it a bit iffier.

> > Could do the clone if not _is_repo() then, and just call _is_repo() in _cloned().
> 
> Given that, I'm not sure what you're asking for here.

Yeah, think they could be merged as well.

Reply:
> Where's the special logic for submodules supposed to go by the way?

Nowhere. The use case I'm trying to support here is to replicate the functionality provided by submodules entirely in the manifest -- i.e., name the subprojects and their revisions so that 'west fetch' puts them in the right places.